Frameworks/GLKit/GLKTexture.mm

```diff
@@ -277,7 +279,8 @@ @implementation GLKTextureLoader {
 */
 + (GLKTextureInfo*)textureWithContentsOfFile:(NSString*)fname options:(NSDictionary*)opts error:(NSError**)err {
     CGDataProviderRef provider = CGDataProviderCreateWithFilename([fname UTF8String]);
-    CGImageRef img = CGImageCreateWithPNGDataProvider(provider, NULL, NO, kCGRenderingIntentDefault);
+    unsigned char* dataBytes = static_cast<unsigned char*>(const_cast<void*>(_CGDataProviderGetData(provider)));
+    CGImageRef img = _CGImageGetImageFromData(dataBytes, _CGDataProviderGetSize(provider));
```

Comment:
Nit: I would add a _CGImageGetImageFromDataProvider() instead of these two lines.